mesop/components/uploader/uploader.py

```diff
@@ -44,24 +46,64 @@ def map_upload_event(event, key):
 register_event_mapper(UploadEvent, map_upload_event)
 
 
-@register_native_component
+@component
 def uploader(
   *,
   label: str,
   accepted_file_types: Sequence[str] | None = None,
   key: str | None = None,
   on_upload: Callable[[UploadEvent], Any] | None = None,
-  type: Literal["raised", "flat", "stroked"] | None = None,
+  type: Literal["raised", "flat", "stroked", "icon"] | None = None,
```

Comment:
I think you're trying to have parity between uploader and content_uploader API but it doesn't really make sense to have icon as a type for uploader because you can't configure an icon here